### DISCUSSION TITLE
Textbox mask for 1-99 range

### DISCUSSION THREAD

**Discussion by ShehanIshanka** [h2oai/wave#1806] [General]
What is the value needed to provide to `mask` param in `textbox` component for the integer range of 1-99? Providing 99 will enable user to type 0.

**Comment by vopani** [comment]
There is no option for 1-9. It's a custom requirement.
Maybe the mask should support regex as a solution to solving all such custom masks.

**Reply by ShehanIshanka** [reply]
Thank you @vopani for the reply. Yeah, supporting regex would be the proper solution.
cc: @mturoci

**Comment by mturoci** [comment]
While regex support could be done, it would bring UX problems and do more harm than good. For example, if one tries to input 0, the typing will ignore it, making the user think your input is broken because you did not explicitly tell them what's wrong.

I suggest using the mask for its original purpose - keeping the format as it should be, doing the validation server side (since you need to do it on the server anyway), and showing the proper error message if fails.
